src/Expr.h

```diff
@@ -1168,21 +1168,24 @@ class ArithCoerceExpr final : public UnaryExpr {
 class RecordCoerceExpr final : public UnaryExpr {
 public:
 	RecordCoerceExpr(ExprPtr op, RecordTypePtr r);
-	~RecordCoerceExpr() override;
 
 	// Optimization-related:
 	ExprPtr Duplicate() override;
 
+	const std::vector<int>& Map() const	{ return map; }
+
 protected:
 	ValPtr InitVal(const zeek::Type* t, ValPtr aggr) const override;
 	ValPtr Fold(Val* v) const override;
 
 	// For each super-record slot, gives subrecord slot with which to
 	// fill it.
-	int* map;
-	int map_size;	// equivalent to Type()->AsRecordType()->NumFields()
+	std::vector<int> map;
 };
 
+extern RecordValPtr coerce_to_record(RecordTypePtr rt, Val* v,
```

Comment:
Does this need to be in the global namespace?

Reply:
The compiler needs to generate code that'll call it.  I guess it could be a static member function, though for my taste that seems a bit weird since it would tie its definition to the interpreter when it's really meant to be stand-alone functionality.